.travis.yml

```diff
@@ -120,7 +114,7 @@ install:
         <linkflags>-static-libgcc ;" >> ~/user-config.jam;
     fi;'
   - 'if [ $clang_tidy == "1" ]; then
-      echo "using clang_tidy : : clang-tidy "-checks=-clang-analyzer-core.*" : <cxxflags>-std=c++11 <cxxflags>-stdlib=libc++ <linkflags>-stdlib=libc++ ;" >> ~/user-config.jam;
+      echo "using clang_tidy : : clang-tidy "-checks=-clang-analyzer-core.*" : <cxxflags>-std=c++11 <cxxflags>-I/usr/local/clang-7.0.0/include/c++/v1 <cxxflags>-stdlib=libc++ <linkflags>-stdlib=libc++ ;" >> ~/user-config.jam;
```

Comment:
that one was a little difficult to figure out :), as you can see above `libc++-dev` was removed as a dependency, and yes, it happens that the one included in xenial is old and buggy, to the point that the `llvm` tools included by travis (`7.0.0`) surface errors in it, and due to the nature of templates, those errors are marked in `libtorrent` sources (not able to filter them). But results that the whole `clang/llvm` toolchain is installed in `/usr/local/clang-7.0.0/`, and since `clang-tidy` only cares about source, just adding that included path do the trick.